File: Sorting/NobleInteger2.java
import java.util.Arrays;

class NobleInteger2 {
    int nobleInteger2(int[] A) {
        int n = A.length;
        Arrays.sort(A);
        int c = 0, ans = 0;
        if (A[0] == 0)
            ans = 1;
        for (int i = 1; i < n; i++) {
            if (A[i] != A[i - 1])
                c = i;

            if (A[i] == c)
                ans++;
        }

        return ans;
    }

    public static void main(String[] args) {
        int[] arr = { -10, 1, 1, 3, 100 };
        NobleInteger2 n = new NobleInteger2();
        System.out.println(n.nobleInteger2(arr));
    }
}

File: Sorting/nobleInteger2.py
def nobleInteger2(A):
    n = len(A)
    A.sort()
    ans = 0
    c = 0
    if A[0] == 0:
        ans = 1
    for i in range(1, n):
        if A[i] != A[i - 1]:
            c = i

        if A[i] == c:
            ans += 1

    return ans


print(nobleInteger2([-10, 1, 1, 3, 100]))
print(nobleInteger2([0, 0, 1, 1, 2]))
